### ISSUE TITLE
bug: Join doesnt seem to render correctly

### ISSUE THREAD

**Issue by DR-DinoMight** [saadeghi/daisyui#2208]
### What version of daisyUI are you using?

3.5.0 +

### Describe your issue

First of all, thank you for the repo you're helping this backend dev actually make nice UIs.

I seem to be having issues using the `join` class, where the buttons aren't being joined up. At first thought this was user error but then I tried the vite example on Stackblitz and changed the button out for one of the snippets of the site and it is still occurring.

![Screenshot 2023-08-02 092729](https://github.com/saadeghi/daisyui/assets/1541665/ddb0bd9c-955f-40cc-a300-6afbd8bb6928)

### What browsers are you seeing the problem on?

All browsers

### Reproduction URL

https://stackblitz.com/edit/daisyui-vite-ttwvxu

**Comment by NikarashiHatsu** [comment]
There might be some misconfiguration in the `tailwind.config.js` or the Stackblitz itself, I'm not sure what went wrong.

On Tailwind Play, it is working as intended https://play.tailwindcss.com/BoeTKKyOuR

**Comment by NikarashiHatsu** [comment]
<img width="275" alt="Screenshot 2023-08-02 at 16 19 04" src="https://github.com/saadeghi/daisyui/assets/43662951/c9a9176f-16ff-46b1-8fd6-08db80d6b50b">

I think I figure it out, you are using the DaisyUI v2, meanwhile `.join` class is introduced on v3. Similar component is using the `Button Group` which is documented on this page https://v2.daisyui.com/components/button-group/

Example usage as follows:
```html
<div class="btn-group">
  <input type="radio" name="options" data-title="1" class="btn" />
  <input type="radio" name="options" data-title="2" class="btn" checked />
  <input type="radio" name="options" data-title="3" class="btn" />
  <input type="radio" name="options" data-title="4" class="btn" />
</div>
```

Example on Stackblitz
https://stackblitz.com/edit/daisyui-vite-lmmfa8?file=tailwind.config.js

**Comment by DR-DinoMight** [comment]
I think you're right, I just presumed the Vite example was running the up-to-date version (might be a separate issue to get those up to date if I have time I may update and put a PR).

Loaded with this knowledge, I was on 3.5.0 so I have just deleted my `node_modules` and re-installed and it also now seems to be working locally now to 😄 

So I think this can be closed, as I think the initial issue has been solved, however, if it's warranted/okay I may look at raising a new issue to look at updating the vite example to use the latest version?

Thank you again all!
